src/main/kotlin/io/emeraldpay/dshackle/startup/ConfiguredUpstreams.kt

```diff
@@ -167,13 +178,26 @@ open class ConfiguredUpstreams(
             return null
         }
         val urls = ArrayList<URI>()
-        val connectorFactory = buildEthereumConnectorFactory(config.id!!, execution, chain, urls, NoChoiceWithPriorityForkChoice(conn.upstreamRating), BlockValidator.ALWAYS_VALID)
+        val connectorFactory = buildEthereumConnectorFactory(
+            config.id!!,
+            execution,
+            chain,
+            urls,
+            NoChoiceWithPriorityForkChoice(conn.upstreamRating),
+            BlockValidator.ALWAYS_VALID
+        )
         val methods = buildMethods(config, chain)
         if (connectorFactory == null) {
             return null
         }
+
+        val hashUrl = conn.execution!!.let {
```

Comment:
After thinking about it, let's add some configuration parameter for upstream. Because this would be more predictable. But keep this logic if hash parameter is not set. Let's call this parameter something like node-id or smith